src/components/questionsSetScreen/questionsSetScreen.container.js

```diff
@@ -1,68 +1,73 @@
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
 import { QuestionsSetScreenComponent } from "./questionsSetScreen.component";
 import { useRoute } from "@react-navigation/native";
 import { StyleSheet, Text, TouchableOpacity } from "react-native";
 import { ChooseMode } from "./components/chooseMode";
+import { useApi } from "../../utils/api";
 
 export const QuestionsSetScreenContainer = ({navigation}) => {
   const route = useRoute();
-  const { headerTitle, titles, mainColor, textColor, img, category, prevScreen } = route.params;
+  const { headerTitle, categoryTopics, mainColor, textColor, img, category, prevScreen } = route.params;
   const [isChooseModeDialog, setIsChooseModeDialog] = useState(false);
-  const [currentTopic, setCurrentTopic] = useState();
-  const [item, setItem] = useState();
-  const getTitle = (item) => {
-    if (prevScreen === 'CategoryCard') {
-      return item.item.name
-    } else {
-      return 'Set - ' + item.index
-    }
-  }
+  const [questionsSets, setQuestionsSets] = useState([]);
+  const [chosenQuestionsSet, setChosenQuestionsSet] = useState();
+  const api = useApi('http://localhost:1339/');
 
-  const renderItem = (item) => {
-    const title = getTitle(item);
+  useEffect(() => {
+    api.questionsSets().then( (questionSets) => {
+     return setQuestionsSets(questionSets.data);
+    });
+  }, []);
+
+  const renderItem = (categoryTopic) => {
+    const title = categoryTopic.item.name;
     return (
       <TouchableOpacity style={[styles.item, {backgroundColor: mainColor}]} onPress={() => {
-        navigateToNextScreen(item)
+        navigateToNextScreen(categoryTopic)
       }}>
         <Text style={{color: textColor}}>{title}</Text>
       </TouchableOpacity>)
   }
 
-  const getTopicsName = () => {
-    return titles.map((element) => element.name);
+  const getQuestionsSetsNames = (categoryTopic) => {
```

Comment:
you don't get questions' sets' names, you filter them by name. Rename function accordingly